freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -230,4 +247,14 @@ private EnvironmentAccessChecker createEnvironmentAccessChecker(String environme
         requireNonNull(environmentCrn, "environmentCrn is null");
         return new EnvironmentAccessChecker(grpcUmsClient, environmentCrn);
     }
+
+    private Crn getCrn(WorkloadAdministrationGroup wag) {
+        Crn resourceCrn = null;
+        try {
+            resourceCrn = Crn.fromString(wag.getResource());
+        } catch (Exception e) {
+            LOGGER.debug("Invalid resource is assigned to workload admin group: {}", e.getMessage());
+        }
+        return resourceCrn;
+    }
```

Comment:
So we have these utility functions in Crns.java and Crn.java to do things like Crns.safeFromString (that can't return null) and Crns.forTypr(String, Service, ResourceType) that ensures that a crn belongs to a service and is of a specific resource type. It may be useful to bring that in here.

Reply:
Thanks. I'll bring that one in along with some tests.